src/tronls.jl

```diff
@@ -64,32 +66,90 @@ stats = tron(nls)
 # output
 "Execution stats: first-order stationary"
 ```
+
+```jldoctest; output = false
+using JSOSolvers, ADNLPModels
+F(x) = [x[1] - 1.0; 10 * (x[2] - x[1]^2)]
+x0 = [-1.2; 1.0]
+nls = ADNLSModel(F, x0, 2, zeros(2), 0.5 * ones(2))
+solver = TronSolverNLS(nls)
+stats = solve!(solver, nls)
+# output
+"Execution stats: first-order stationary"
+```
 """
-function tron(
+mutable struct TronSolverNLS{T, V <: AbstractVector{T}, Op <: AbstractLinearOperator{T}} <: AbstractOptimizationSolver
```

Comment:
```suggestion
mutable struct TronSolverNLS{T <: AbstractFloat, V <: AbstractVector{T}, Op <: AbstractLinearOperator{T}} <: AbstractOptimizationSolver
```

Reply:
What is the benefit of adding this?

Reply:
We know what kind of problems we can solve with it.
Is your code working with functions `F(x)` that take as input complex numbers?
`||F(x)||` is still a real number.

Reply:
No idea, this has been here for a while. I would keep the code the way it is, without this just in case someone was using it for something else than `Float`s.